make check-generate is flaky

**How to categorize this issue?**
<!--
Please select area, kind, and priority for this issue. This helps the community categorizing it.
Replace below TODOs or exchange the existing identifiers with those that fit best in your opinion.
If multiple identifiers make sense you can also state the commands multiple times, e.g.
  /area control-plane
  /area auto-scaling
  ...

"/area" identifiers:     audit-logging|auto-scaling|backup|certification|control-plane-migration|control-plane|cost|delivery|dev-productivity|disaster-recovery|documentation|high-availability|logging|metering|monitoring|networking|open-source|ops-productivity|os|performance|quality|robustness|scalability|security|storage|testing|usability|user-management
"/kind" identifiers:     api-change|bug|cleanup|discussion|enhancement|epic|impediment|poc|post-mortem|question|regression|task|technical-debt|test
"/priority" identifiers: 1|2|3|4|5 (ordered from greatest to least)
-->
/area dev-productivity
/kind flake

**What happened**:

From time to time, `make check-generate` fails in pipeline jobs with errors like these:

https://concourse.ci-2.gardener.cloud/teams/gardener/pipelines/gardener-master/jobs/master-head-update-job/builds/521#L6110250c:518
```
# github.com/gardener/gardener/pkg/gardenlet/apis/config/v1alpha1
../../../../../pkg/gardenlet/apis/config/v1alpha1/defaults.go:32:9: undefined: RegisterDefaults
../../../../../pkg/gardenlet/apis/config/v1alpha1/register.go:51:3: cannot use &GardenletConfiguration{} (type *GardenletConfiguration) as type runtime.Object in argument to scheme.AddKnownTypes:
	*GardenletConfiguration does not implement runtime.Object (missing DeepCopyObject method)
# github.com/gardener/gardener/pkg/gardenlet/apis/config/v1alpha1
../../../../pkg/gardenlet/apis/config/v1alpha1/defaults.go:32:9: undefined: RegisterDefaults
../../../../pkg/gardenlet/apis/config/v1alpha1/register.go:51:3: cannot use &GardenletConfiguration{} (type *GardenletConfiguration) as type runtime.Object in argument to scheme.AddKnownTypes:
	*GardenletConfiguration does not implement runtime.Object (missing DeepCopyObject method)
prog.go:12:2: cannot find package "github.com/golang/mock/mockgen/model" in any of:
	/usr/local/go/src/github.com/golang/mock/mockgen/model (from $GOROOT)
	/go/src/github.com/golang/mock/mockgen/model (from $GOPATH)
2021/08/08 07:08:45 Loading input failed: exit status 1
extensions/pkg/util/mock/doc.go:15: running "mockgen": exit status 1
```

https://concourse.ci-2.gardener.cloud/teams/gardener/pipelines/gardener-master/jobs/master-head-update-job/builds/512#L61089570:382
```
Generating API groups for pkg/apis/operations
Generating deepcopy funcs
Generating conversions
F0804 07:31:41.659377   11253 deepcopy.go:813] Hit an unsupported type invalid type for []invalid type
goroutine 1 [running]:
github.com/gardener/gardener/vendor/k8s.io/klog/v2.stacks(0xc000010001, 0xc00c551680, 0x67, 0xb7)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/klog/v2/klog.go:1026 +0xb9
github.com/gardener/gardener/vendor/k8s.io/klog/v2.(*loggingT).output(0x971840, 0xc000000003, 0x0, 0x0, 0xc00c4dfc00, 0x0, 0x836217, 0xb, 0x32d, 0x0)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/klog/v2/klog.go:975 +0x1e5
github.com/gardener/gardener/vendor/k8s.io/klog/v2.(*loggingT).printf(0x971840, 0xc000000003, 0x0, 0x0, 0x0, 0x0, 0x77148b, 0x21, 0xc00c635820, 0x2, ...)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/klog/v2/klog.go:753 +0x19a
github.com/gardener/gardener/vendor/k8s.io/klog/v2.Fatalf(...)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/klog/v2/klog.go:1514
github.com/gardener/gardener/vendor/k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).doSlice(0xc00c4e1600, 0xc00c3ce840, 0xc00c57f090)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:813 +0x5a6
github.com/gardener/gardener/vendor/k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).generateFor(0xc00c4e1600, 0xc00c3ce840, 0xc00c57f090)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:695 +0xcf
github.com/gardener/gardener/vendor/k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).doStruct(0xc00c4e1600, 0xc00c3bca80, 0xc00c57f090)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:865 +0x4ed
github.com/gardener/gardener/vendor/k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).generateFor(0xc00c4e1600, 0xc00c3bca80, 0xc00c57f090)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:695 +0xcf
github.com/gardener/gardener/vendor/k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).GenerateType(0xc00c4e1600, 0xc00c4df180, 0xc00c3bca80, 0x7c2538, 0xc00c4d7760, 0x0, 0x0)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:608 +0xf98
github.com/gardener/gardener/vendor/k8s.io/gengo/generator.(*Context).executeBody(0xc00c4df180, 0x7c2398, 0xc0000d9470, 0x7cad48, 0xc00c4e1600, 0x380, 0xc00b97fbc0)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/gengo/generator/execute.go:306 +0x122
github.com/gardener/gardener/vendor/k8s.io/gengo/generator.(*Context).ExecutePackage(0xc0000cbce0, 0xc000016736, 0x7, 0x7c94b0, 0xc00c4b6d80, 0x0, 0x0)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/gengo/generator/execute.go:267 +0xb98
github.com/gardener/gardener/vendor/k8s.io/gengo/generator.(*Context).ExecutePackages(0xc0000cbce0, 0xc000016736, 0x7, 0xc00c4d74c0, 0x2, 0x2, 0x0, 0x28)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/gengo/generator/execute.go:51 +0xc8
github.com/gardener/gardener/vendor/k8s.io/gengo/args.(*GeneratorArgs).Execute(0xc0000bb0e0, 0xc00b97fe38, 0x74c53c, 0x6, 0x77a578, 0x0, 0x0)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/gengo/args/args.go:207 +0x1fa
main.main()
	/go/src/github.com/gardener/gardener/vendor/k8s.io/code-generator/cmd/deepcopy-gen/main.go:77 +0x4cb
```

**What you expected to happen**:

`make check-generate` not to flake.

**How to reproduce it (as minimally and precisely as possible)**:

Not clear yet.

--- a/Makefile
+++ b/Makefile
@@ -189,7 +189,7 @@ check: $(GOIMPORTS) $(GOLANGCI_LINT) $(HELM)
 .PHONY: generate
 generate: $(CONTROLLER_GEN) $(GEN_CRD_API_REFERENCE_DOCS) $(GOIMPORTS) $(GO_TO_PROTOBUF) $(MOCKGEN) $(OPENAPI_GEN) $(PROTOC_GEN_GOGO) $(YAML2JSON)
 	@hack/update-protobuf.sh
-	@hack/update-codegen.sh --parallel
+	@hack/update-codegen.sh
 	@hack/generate-parallel.sh charts cmd example extensions pkg plugin landscaper test
 	@hack/generate-monitoring-docs.sh
 

--- a/hack/update-codegen.sh
+++ b/hack/update-codegen.sh
@@ -381,7 +381,7 @@ if [[ $# -gt 0 && "$1" == "--parallel" ]]; then
     scheduler_groups \
     gardenlet_groups \
     shoottolerationrestriction_groups \
-    landscapergardenlet_groups
+    landscapergardenlet_groups \
     landscapercontrolplane_groups
 else
   authentication_groups

